src/frequenz/sdk/timeseries/battery_pool/_result_types.py

```diff
@@ -0,0 +1,181 @@
+class CapacityMetrics:
+    """Capacity metrics"""
+
+    total_capacity: float
+    """Total batteries capacity.
+
+    Calculated with the formula:
+    ```
+    working_batteries: Set[BatteryData] # working batteries from the battery pool
+    total_capacity = sum(battery.capacity for battery in working_batteries)
+    ```
+    """
+
+    capacity_upper_bound: float
+    """Max capacity that can be stored in the batteries.
+
+    Calculated with the formula:
+    ```
+    working_batteries: Set[BatteryData] # working batteries from the battery pool
+    capacity_upper_bound = sum(
+        battery.capacity * battery.soc_upper_bound for battery in working_batteries)
+    ```
+    """
+
+    capacity_lower_bound: float
+    """Min capacity that has to be stored in the batteries
+
+    Calculated with the formula:
+    ```
+    working_batteries: Set[BatteryData] # working batteries from the battery pool
+    capacity_lower_bound = sum(
+        battery.capacity * battery.soc_lower_bound for battery in working_batteries)
+    ```
+    """
+
+
+@dataclass
+class SoCMetrics:
+    "Soc metrics"
+
+    average_soc: float
+    """Average soc.
+
+    Calculated with the formula:
+    ```
+    working_batteries: Set[BatteryData] # working batteries from the battery pool
+
+    used_capacity = sum(battery.capacity * battery.soc for battery in working_batteries)
+    total_capacity = sum(battery.capacity for battery in working_batteries)
+    average_soc = used_capacity/total_capacity
+    ```
+    """
+
+    soc_lower_bound: float
+    """Lower soc bound averaged in all working batteries.
+
+    Calculated with the formula:
+    ```
+    working_batteries: Set[BatteryData] # working batteries from the battery pool
+
+    capacity_lower_bound = sum(
+        battery.capacity * battery.soc_lower_bound for battery in working_batteries)
+    total_capacity = sum(battery.capacity for battery in working_batteries)
+    soc_lower_bound = capacity_lower_bound/total_capacity
+    ```
+    """
+
+    soc_upper_bound: float
+    """Upper soc bound averaged in all working batteries.
+
+    Calculated with the formula:
+    ```
+    working_batteries: Set[BatteryData] # working batteries from the battery pool
+
+    capacity_upper_bound = sum(
+        battery.capacity * battery.soc_upper_bound for battery in working_batteries)
+    total_capacity = sum(battery.capacity for battery in working_batteries)
+    soc_lower_bound = capacity_upper_bound/total_capacity
+    ```
+    """
+
+
+# Soc and capacity are different kind of metrics.
+# It makes sens to separate them with `_`.
+@dataclass
+class SoC_Capacity_Metrics:  # pylint: disable=invalid-name
+    """SoC and capacity metrics"""
+
+    soc_metrics: SoCMetrics
+    """Sec metrics."""
+
+    capacity_metrics: CapacityMetrics
+    """Capacity metrics."""
+
+
+# Soc and capacity are different kind of metrics.
+# It makes sens to separate them with `_`.
+@dataclass
+class SoC_Capacity:  # pylint: disable=invalid-name
```

Comment:
Same here. Maybe `SocCapacityCombined`?

Reply:
I separated it as we discussed on the meeting